python/ray/autoscaler/_private/placement_group_load_util.py

```diff
@@ -0,0 +1,33 @@
+from dataclasses import dataclass
+from numbers import Number
+from typing import Dict, List
+
+STRICT_PACK = 1
+STRICT_SPREAD = 2
+PACK = 3
+SPREAD = 4
+_strat_enum_to_string = ["PACK", "SPREAD", "STRICT_PACK", "STRICT_SPREAD"]
+
+
+def strategy_enum_to_string(strat_pb):
+    assert strat_pb < len(_strat_enum_to_string)
+    return _strat_enum_to_string[strat_pb]
+
+
+PENDING = 0
+CREATED = 1
+REMOVED = 2
+RESCHEDULING = 3
+_state_enum_to_string = ["PENDING", "CREATED", "REMOVED", "RESCHEDULING"]
+
+
+def state_enum_to_string(state_pb):
+    assert state_pb < len(_state_enum_to_string)
+    return _state_enum_to_string[state_pb]
+
+
+@dataclass
```

Comment:
I think we don't need this file if we directly use grpc generated code? In this way, we can avoid adding new dataclass dependency as well.

Reply:
hmmmm I mostly did this out of consistency, since we are pretty consistent about converting grpc messages to native python objects for all the other protobufs. Also, I think the dataclass dependency is pretty safe (I'm surprised we didn't already depend on it). It's a built-in module in python 3.7+ so this is really just a dependency on 3.6 only.

Reply:
If you feel strongly about reusing to protobuf, I wouldn't mind doing that.

Reply:
The reason why we don't have the dataclass dependency so far is I think because we strongly avoid adding default dependencies if possible. Dataclass is useful, but not absolutely necessary. Also, at least in the dashboard, I think we just use the grpc object instead of rebuilding python objects (I don't know autoscaler convention).

I just feel like having additional enum that redefines protobuf enum is a little redundant and could be possibly cause inconsistency issues (I saw you added a comment under protobuf, but in reality, no one reads comments). So I prefer to remove this file. (if using grpc object is not too uncomfortable for you).

Reply:
Ok done. Feels a little weird for the resource demand scheduler to know about protobufs, but it's all contained to one method so I'm ok with it.